### PR TITLE
add "clean-build" github workflow

### LINKED ISSUE #38: Github actions: clean-build
Run the clean-build script on every commit push to add visibility over the code's health.
The clean-build script includes `build`, `lint`, and `test`

### PR DESCRIPTION
**Pull request by Kelerchian**
resolves #38

### CODE REVIEW

**Comment by jmg-duarte** [comment]
I'm missing context, could you explain why this is needed?

**Comment by Kelerchian** [comment]
> I'm missing context, could you explain why this is needed?

This is to add visibility to the code's health.

**Review by jmg-duarte — APPROVED** [review]

File: .github/workflows/clean-build.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        node-version: [19.x]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: jetli/wasm-pack-action@v0.4.0
+        with:
+          version: "latest"
```
Comment (rkuhn) [inline .github/workflows/clean-build.yml:18]: CI should use fixed versions for everything as far as possible

**Review by rkuhn — COMMENTED** [review]
Cool, learnt one more thing (wasm-pack-action)! We should ensure that `master` is always perfectly clean, so could you also fix the `any`-related lint errors before merging?

File: machine-runner/src/runner.test.ts [hunk 2]
```diff
@@ -80,6 +80,7 @@ class Runner<
   private sub = Subscription.make<RegisteredEventsFactoriesTuple>()
   public machine
 
+  /* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (rkuhn) [inline machine-runner/src/runner.test.ts:83]: we can leave it as is, or (for tests or internal code) disable this lint for the whole file — as you prefer

**Review by rkuhn — APPROVED** [review]
Looking good! Feel free to address or ignore my final comment.

**Pull request merged**